packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -7,15 +7,50 @@ import "../v2/FuelMessagePortalV2.sol";
 contract FuelMessagePortalV3 is FuelMessagePortalV2 {
     using FuelBlockHeaderLib for FuelBlockHeader;
     using FuelBlockHeaderLiteLib for FuelBlockHeaderLite;
-
+    
+    error AlreadyInitialized();
     error MessageBlacklisted();
-    error WithdrawalsPaused();
     error MessageRelayFailed();
+    error RateLimitExceeded();
+    error WithdrawalsPaused();
 
+    /// @dev The rate limit setter role
+    bytes32 public constant SET_RATE_LIMITER_ROLE = keccak256("SET_RATE_LIMITER_ROLE");
+
+    /// @notice Duration after which rate limit resets.
+    uint256 public immutable rateLimitDuration;
+
+    /// @notice Flag to indicate whether withdrawals are paused or not.
     bool public withdrawalsPaused;
+
     mapping(bytes32 => bool) public messageIsBlacklisted;
 
-    constructor(uint256 _depositLimitGlobal) FuelMessagePortalV2(_depositLimitGlobal) {}
+    /// @notice Amounts already withdrawn this period.
+    uint256 public currentPeriodAmount;
+
+    /// @notice The time at which the current period ends at.
+    uint256 public currentPeriodEnd;
+
+    /// @notice The eth withdrawal limit amount.
+    uint256 public limitAmount;
+
+    constructor(uint256 _depositLimitGlobal, uint256 _rateLimitDuration) FuelMessagePortalV2(_depositLimitGlobal) {
+        rateLimitDuration = _rateLimitDuration;
+    }
+
+    function initialize(FuelChainState) public virtual override {
+        revert AlreadyInitialized();
```

Comment:
I don't think this error is accurate. I 'd put something along the lines of deprecated.